### ISSUE TITLE
Incorrect installation directory for pkg-config files.

### ISSUE THREAD

**Issue by dkosmari** [SFML/SFML#2778]
### Prerequisite Checklist

- [X] I searched for [existing issues](https://github.com/search?q=repo%3ASFML%2FSFML&type=issues) to prevent duplicates
- [X] I searched for [existing discussions on the forum](https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fen.sfml-dev.org) to prevent duplicates
- [X] I am here to report an issue and not to just ask a question or look for help (use the [forum](https://en.sfml-dev.org/forums/index.php#c3) or [Discord](https://discord.gg/nr4X7Fh) instead)

### Describe your issue here

The pkg-config scripts (`tools/pkg-config/*.pc`) are always installed in the wrong location. This led to all Linux distros trying different tricks to override the wrong installation.

This happens because `SFML_PKGCONFIG_DIR` is not being expanded correctly.
In `Config.cmake`:
```cmake
set(SFML_PKGCONFIG_DIR "/${CMAKE_INSTALL_LIBDIR}/pkgconfig")
```
After the **first** `cmake` run, this variable expands to
```
//pkgconfig
```
After the **second** `cmake` run, it changes to:
```
/lib64/pkgconfig
```
It appears that `CMAKE_INSTALL_LIBDIR` is set AFTER it's used to set `SFML_PKGCONFIG_DIR`.
I think the problem is caused by the wrong order of inclusion in `CMakeLists.txt`:

```cmake
# include the configuration file
include(cmake/Config.cmake)

# we use the paths from the cmake GNUInstallDirs module as defaults
# you can override these if you like
# https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html
include(GNUInstallDirs)
```

If I change it to include `GNUInstallDirs` before `cmake/Config.cmake`, the value seems to be set correctly.

### Your Environment

- OS / distro / window manager: Fedora 39, Mageia 9, Ubuntu Mantic, Debian Bookworm, Arch Linux (latest stable releases of all of them)
- SFML version: 2.6.1
- Compiler / toolchain: GCC
- Special compiler / CMake flags: none

### Steps to reproduce

1. Edit `cmake/Config.cmake`, appending these lines at the bottom:
```cmake
message(STATUS ":::: CMAKE_INSTALL_LIBDIR = ${CMAKE_INSTALL_LIBDIR}")
message(STATUS ":::: SFML_PKGCONFIG_DIR = ${SFML_PKGCONFIG_DIR}")
```
2. `mkdir build`
3. `cd build`
4. `cmake ..`
5. `cmake ..`

### Expected behavior

It should always print
```
-- :::: CMAKE_INSTALL_LIBDIR = lib64
-- :::: SFML_PKGCONFIG_DIR = /lib64/pkgconfig
```
each time `cmake` is run.

### Actual behavior

First `cmake` run prints:
```
-- :::: CMAKE_INSTALL_LIBDIR = 
-- :::: SFML_PKGCONFIG_DIR = //pkgconfig
```
The second run prints:
```
-- :::: CMAKE_INSTALL_LIBDIR = lib64
-- :::: SFML_PKGCONFIG_DIR = /lib64/pkgconfig
```

**Comment by ChrisThrasher** [comment]
> It appears that CMAKE_INSTALL_LIBDIR is set AFTER it's used to set SFML_PKGCONFIG_DIR.

This does appear to be the case. The cache variables added by the GNUInstallDirs module are used in Config.cmake which is included before we include the GNUInstallDirs module.

Can you try out this branch? https://github.com/SFML/SFML/compare/master...fix_module_inclusion_order

**Comment by dkosmari** [comment]
Tested both a native build and a mingw32 cross compilation, on Fedora 39.
The `SFML_PKGCONFIG_INSTALL_PREFIX` variable now contains the correct path, and the `.pc` files are installed in the correct location.

**Comment by ChrisThrasher** [comment]
I submitted a PR to fix this

https://github.com/SFML/SFML/pull/2779

**Comment by dkosmari** [comment]
Please also backport it to the 2.6.x branch, then maybe all the distros won't have to workaround this bug.

**Comment by ChrisThrasher** [comment]
I restarted to 2.6.x. Please approve the PR if you think it's good to go.

**Comment by ChrisThrasher** [comment]
Fixed by #2779
